.github/workflows/hrm-ecs-export.yml

```diff
@@ -0,0 +1,165 @@
+          echo = security_group=$task_definition | jq -r '.networkMode == "awsvpc" | if . then .networkConfiguration.awsvpcConfiguration.securityGroups | join(",") else empty end' >> $GITHUB_ENV
+      
+      # Command to trigger the execution of the ECS task. 
+      - name: Trigger ECS Task
+        id: ecs-task
+        run: |
+          aws ecs run-task \
+            --cluster ${{ env.ECR_REPOSITORY }}-task \
+            --launch-type FARGATE \
+            --task-definition ${{ env.task_definition }} \
+            --network-configuration "awsvpcConfiguration={subnets=['env.subnets'],securityGroups=['env.security_group'],assignPublicIp='ENABLED'}" \
+            --overrides '{
+              "containerOverrides": [
+                {
+                  "name": "${{ env.ECR_REPOSITORY }}-scheduled-task",
+                  "command": [
+                    "npm", 
+                    "run", 
+                    "start:khp-data-pull",
+                    "${{ github.event.inputs.start-date }}",
+                    "${{ github.event.inputs.end-date }}"
+                  ]
+                }
+              ]
+            }'
+
+      # Ensures that the workflow doesn't proceed until the ECS task has finished 
+      # executing. This step prevents any potential race conditions or incorrect 
+      # assumptions about the task's status.
+      - name: Wait for ECS Task Completion
+        id: ecs-task-status
+        run: |
+          aws ecs wait tasks-stopped \
+            --cluster ${{ env.ECR_REPOSITORY }}-task \
+            --tasks ${{ steps.ecs-task.outputs.task-id }}
+
+      #  Retrieve the output of the ECS task that was triggered and extract 
+      # the desired information from it. 
+      - name: Get ECS Task Output
+        id: ecs-task-output
+        run: |
+          aws ecs describe-tasks \
+            --cluster ${{ env.ECR_REPOSITORY }}-task \
+            --tasks ${{ steps.ecs-task.outputs.task-id }} \
+            --query "tasks[0].containers[0].command[1:3]" \
+            --output text
+
+      # Extracts the relevant information from the ECS task output 
+      # allowing us to process the task output and extract specific data that is needed
+      - name: Extract S3 Bucket Info
+        id: s3-bucket-info
+        run: |
+          start_date=$(echo "${{ steps.ecs-task-output.outputs.stdout }}" | awk '{print $1}')
+          end_date=$(echo "${{ steps.ecs-task-output.outputs.stdout }}" | awk '{print $2}')
+
+          echo "::set-output name=start-date::$start_date"
+          echo "::set-output name=end-date::$end_date"
+
+      # responsible for uploading files to the S3 bucket based on the extracted 
+      # start date and end date. 
+      - name: Populate S3 Bucket
+        if: ${{ steps.ecs-task-status.outputs.tasksStopped == 'true' }}
+        run: |
+          aws s3 cp /path/to/local/files s3://your-bucket-name/path/to/s3/location --recursive
```

Comment:
I think these 2 steps won't be needed. The S3 population will be done by the ECS task.

Reply:
@murilovmachado  - Noted. I have removed both tasks.